### PR TITLE
chore: P4ADEV-787 machine 2 machine

### PR DESCRIPTION
**Pull request by antocalo**
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: design/a2a/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      operationId: postToken
+      security: []
+      parameters:
+        - name: client_id
+          in: query
+          required: true
+          schema:
+            enum: [
+              "piattaforma-unitaria"
+            ]
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: this cannot be more an enum but a free text field

File: design/a2a/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+          schema:
+            enum: [
+              "urn:ietf:params:oauth:grant-type:token-exchange",
+              "client_credentials"
+            ]
+            type: string
+        - name: subject_token
+          description: "ID Token"
+          in: query
+          required: true
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: not more mandatory when grant_type is client_credentials (removing the mandatory requirement, we should check if its presence is checked by java code)

File: design/a2a/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+          schema:
+            type: string
+        - name: subject_issuer
+          description: "Issuer del subject_token"
+          in: query
+          required: true
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: not more mandatory when grant_type is client_credentials (removing the mandatory requirement, we should check if its presence is checked by java code)

File: design/a2a/openapi.yml [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+          schema:
+            type: string
+        - name: subject_token_type
+          in: query
+          required: true
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: not more mandatory when grant_type is client_credentials (removing the mandatory requirement, we should check if its presence is checked by java code)

File: design/a2a/openapi.yml [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+      operationId: logout
+      security: []
+      parameters:
+        - name: client_id
+          in: query
+          required: true
+          schema:
+            enum: [
+              "piattaforma-unitaria"
+            ]
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: not more an enum

File: design/a2a/openapi.yml [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '400':
+          description: Invalid request
+        '401':
+          description: Invalid client_id
+  /am/clients/{organizationIpaCode}:
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: I am not sure that "clients" is a Access Management entity...
I think it should still be an Authentication concept
```suggestion
  /auth/clients/{organizationIpaCode}:
```

File: design/a2a/openapi.yml [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+          description: "The number of the page"
+          schema:
+            type: integer
+            default: 0
+        - name: size
+          in: query
+          description: "Number of items"
+          schema:
+            type: integer
+            default: 10
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: a paged query could be an over-engineered feature?
I don't think we should have such a large number of clients...

File: design/a2a/openapi.yml [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Client'
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: This should return a list of data (if paged, with the detail of the page, but I think that this API could not be paged)

Comment (antonioT90) [reply design/a2a/openapi.yml]: MoreOver, let's use different model to return the Client data:
Client, with just the clientId
ClientDetail, return by the findById, which will return (and decrypt) the clientSecret, which will be mandatory

**Review by antonioT90 — CHANGES_REQUESTED** [review]

**Review by LarissaASLeite — APPROVED** [review]

File: design/a2a/openapi.yml [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                $ref: '#/components/schemas/ClientDTO'
```
Comment (antonioT90) [inline design/a2a/openapi.yml]: This API should not return the clientSecret

**Review by antonioT90 — CHANGES_REQUESTED** [review]

**Review by antonioT90 — APPROVED** [review]

**Pull request merged**